janus/__init__.py

```diff
@@ -223,14 +224,16 @@ def f() -> None:
             with self._sync_mutex:
                 self._sync_not_empty.notify()
 
-        self._loop.run_in_executor(None, f)
+        asyncio.ensure_future(anyio.to_thread.run_sync(f, *[]))
+        # self._loop.run_in_executor(None, f)
 
     def _notify_sync_not_full(self) -> None:
         def f() -> None:
             with self._sync_mutex:
                 self._sync_not_full.notify()
 
-        fut = asyncio.ensure_future(self._loop.run_in_executor(None, f))
+        fut = asyncio.ensure_future(anyio.to_thread.run_sync(f, *[]))
```

Comment:
I do not think there is any interest to support anyio between maintainers, this is asyncio library after all. I do not have experience with it, but may be `anyio` can provide custom `executor` so `self._loop.run_in_executor` can call `anyio.to_thread.run_sync(f, *[])` ? Then there is not need to change library itself much, other then may be provide executor for the loop.